src/pages/index.js

```diff
@@ -65,35 +86,128 @@ profileEditButton.addEventListener("click", () => {
+    handleCardImageClick,
+    myUserId: userId,
+    handleDeleteCard,
+    handleLikeButton: (cardId) => {
+      if (card.isLiked()) {
+        api
+          .removeLike(cardId)
+          .then((data) => {
+            card.updateLikes(data.likes);
+          })
+          .catch((err) => {
+            console.log(err);
+          });
+      } else {
+        api
+          .addLike(cardId)
+          .then((data) => {
+            card.updateLikes(data.likes);
+          })
+          .catch((err) => {
+            console.log(err);
+          });
+      }
+    },
+  });
+  return card.getCardElement();
+}
+let cardList;
+api
+  .getInitialCards()
+  .then((initialCards) => {
+    cardList = new Section(
+      {
+        items: initialCards,
+        renderer: (data) => {
+          const newCard = renderCard(data);
+          cardList.addItem(newCard);
+        },
+      },
+      cardsWrap
+    );
+    cardList.renderItems();
+  })
+  .catch((err) => console.log(err));
+
+let userId;
+api
+  .getUserInfo()
+  .then((formData) => {
+    userId = formData._id;
+    userInfo.setProfileAvatar(formData.avatar);
+    userInfo.setUserInfo(formData.name, formData.about);
+  })
+  .catch((err) => console.log(err));
+
+function handleEditProfileSubmit({ nameInfo, jobInfo }) {
+  api
+    .changeUserInfo(nameInfo, jobInfo)
+    .then(() => {
+      userInfo.setUserInfo(nameInfo, jobInfo);
+      editPopupForm.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 }
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
-  },
-  cardsWrap
-);
-
-section.renderItems();
-
-// Handlers/
-
-function handleEditProfileSubmit(formData) {
-  const { nameInfo, jobInfo } = formData;
-  userInfo.setUserInfo(nameInfo, jobInfo);
-  editPopupForm.close();
+function handleAddCardFormSubmit({ name, link }) {
+  api
+    .addCard(name, link)
+    .then((cardElement) => {
+      const newCard = renderCard(cardElement);
+      cardList.prependItem(newCard);
+      addCardPopupForm.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+//Delete Card
+function handleDeleteCard(cardId) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.renderLoading(true);
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        this.deleteConfirmClick();
+        deleteCardPopup.close();
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => {
+        deleteCardPopup.renderLoading(false);
+      });
+  });
 }
 
-function handleAddCardFormSubmit(inputValues) {
-  const { name, link } = inputValues;
-  renderCard({ name, link });
-  addCardPopupForm.close();
+//Avatar
+function handleAvatarFormSubmit(data) {
+  api
+    .changeUserAvatar(data.avatarUrl)
+    .then((formData) => {
+      userInfo.setProfileAvatar(formData.avatar);
+      avatarProfilePopupForm.close();
+    })
+    .catch((err) => {
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors